blogs/2022-06-28-tips-kafka-streams-distributed.md

```diff
@@ -11,18 +11,18 @@ image: /blogs/2022-06-28-tips-kafka-streams-distributed.jpg
 draft: true
 ---
 
-While building [Kestra](https://github.com/kestra-io/kestra), an open-source data orchestration and scheduling platform, we decided to use Kafka as the main database to build a scalable architecture. Since Kafka allows gigabytes of throughput, we are confident to have a fully scalable solution, but we need to adapt to Kafka.
+The distributed event store and streamlining platform [Apache Kafka](https://kafka.apache.org/) have many advantages. Think of high throughput, low latency, durability, high concurrency, scalability, and fault tolerance. So, it's not surprising that when we built [Kestra](https://github.com/kestra-io/kestra), an open-source data orchestration and scheduling platform, we decided to use Kafka as the central database to build a scalable architecture. Since Kafka supports high throughput, which allows it to handle high-volume data at equally high velocity, we are confident of a fully scalable solution. However, we need to adapt to Kafka.
```

Comment:
@tchiotludo What do we mean by "we need to adapt to Kafka."?

Reply:
Kafka have some restriction, since it was not a database, so we need to deal with the constraint and adapt the code to make it work with Kafka